plugin/plugin.go

```diff
@@ -55,25 +70,29 @@ func (p *plugin) Validate(ctx context.Context, req *validator.Request) error {
 		// check if user is privilged to promote to any env
 		if stringInSlice(req.Build.Trigger, p.privilegedUsers) {
 			logrus.Debugf(
-				"User %s has been authorized to %s to/on %s env as a privileged user",
-				req.Build.Trigger, req.Build.Event, req.Build.Deploy,
+				"User %s has been authorized to %s to/on %s env in %s repo as a privileged user",
+				req.Build.Trigger, req.Build.Event, req.Build.Deploy, req.Repo.Name,
 			)
 			return nil
 		}
 
-		// check if user has any per-env permission
+		// check if user has any per-env[repo] permission
 		if allowedEnvs, userHasPermissions := p.userPermissions[req.Build.Trigger]; userHasPermissions {
 			// check if user is allowed to promote to a requested env
-			if stringInSlice(req.Build.Deploy, allowedEnvs) {
-				logrus.Debugf(
-					"User %s has been authorized to %s to/on %s env according to user level permissions",
-					req.Build.Trigger, req.Build.Event, req.Build.Deploy,
-				)
-				return nil
+
+			for env, repos := range allowedEnvs {
+				if env == req.Build.Deploy && stringInSlice(req.Repo.Name, repos) {
+					logrus.Debugf(
```

Comment:
Use fields so the log is more machine parseable. We should actually switch to zap but that's ok for now. Ex: https://github.com/getgoing/drone-amazon-ssm/blob/master/plugin/plugin.go#L56